### PR TITLE
[r,python] Add `show_package_versions` for bug-report use

### PR DESCRIPTION
**Pull request by johnkerl**
## Issue and/or context:

#851 

## Changes:

As on #851

### CODE REVIEW

File: apis/python/src/tiledbsoma/general_utilities.py [hunk 1]
```diff
@@ -38,3 +41,22 @@ def get_storage_engine() -> str:
     [lifecycle: experimental]
     """
     return "tiledb"
+
+
+def show_package_versions() -> None:
+    """
+    Nominal use is for bug reports, so issue filers and issue fixers can be on the same page.
+    """
+    line1 = "tiledbsoma.__version__   " + get_implementation_version()
```
Comment (thetorpedodog) [inline apis/python/src/tiledbsoma/general_utilities.py:50]: why not print these immediately? no need to make vars, build a list, join the list, then print the result.

File: apis/python/src/tiledbsoma/general_utilities.py [hunk 2]
```diff
@@ -38,3 +41,22 @@ def get_storage_engine() -> str:
...
+    line3 = "python__version__        " + ".".join(
+        [
+            str(e)
+            for e in [
+                sys.version_info.major,
+                sys.version_info.minor,
+                sys.version_info.micro,
+            ]
+        ]
+    )
```
Comment (thetorpedodog) [inline apis/python/src/tiledbsoma/general_utilities.py:61]:
```python
".".join(str(v) for v in sys.version_info)
```

File: apis/python/src/tiledbsoma/general_utilities.py (hunk 1 above)
Comment (johnkerl) [reply apis/python/src/tiledbsoma/general_utilities.py:50]: @thetorpedodog d'oh

**Comment by johnkerl** [comment]
@eddelbuettel OK on the R side?

**Review by thetorpedodog — APPROVED** [review]
python looks good

File: apis/r/R/utils-tiledb.R [hunk 3]
```diff
@@ -18,3 +18,14 @@ map_query_layout <- function(layout) {
...
+#' show_package_versions
+#'
+#' @description Prints package information suitable for assisting with bug reports.
+#'
+#' @export
+show_package_versions <- function() {
+    cat("tiledbsoma:   ", toString(utils::packageVersion("tiledbsoma")), "\n")
+    cat("tiledb:       ", toString(utils::packageVersion("tiledb")), "\n")
+    cat("R:            ", R.Version()$version.string, "\n")
+}
```
Comment (eddelbuettel) [inline apis/r/R/utils-tiledb.R:31]: Suggest to change to

```r
#' show_package_versions
#'
#' @description Prints package information suitable for assisting with bug reports.
#'
#' @export
#' @importFrom utils packageVersion
show_package_versions <- function() {
    cat("tiledbsoma:   ", toString(utils::packageVersion("tiledbsoma")), "\n")
    cat("tiledb:       ", toString(utils::packageVersion("tiledb")), "\n")
    cat("tiledb core:  ", tiledb::tiledb_version(TRUE), "\n")
    cat("R:            ", R.version.string, "\n")
}
```

(untested)

**Review by eddelbuettel — CHANGES_REQUESTED** [review]

File: apis/r/R/utils-tiledb.R (hunk 3 above)
Comment (johnkerl) [reply apis/r/R/utils-tiledb.R:33]: @eddelbuettel added in latest commit

@thetorpedodog I added the core-version line to the Python side as well (great idea!)

Comment (eddelbuettel) [reply apis/r/R/utils-tiledb.R:33]: Cool, thanks, still haven't had a chance to run it so not sure we need `toString()`, part of me thinks that formatting should be automagic in the context.   

Oh, and I just checked, we already have `importFrom(utils,packageVersion)` in `NAMESPACE` so my add in what is now line 27 above wasn't needed (but won't hurt).

**Review by eddelbuettel — APPROVED** [review]

File: apis/r/R/utils-tiledb.R (hunk 3 above)
Comment (johnkerl) [reply apis/r/R/utils-tiledb.R:33]: All good @eddelbuettel , I ran it :)

**Pull request merged**